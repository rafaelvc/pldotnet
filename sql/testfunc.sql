CREATE OR REPLACE FUNCTION returnX() RETURNS integer AS $$
return 10;
$$ LANGUAGE pldotnet;
SELECT returnX() = integer '10';

CREATE OR REPLACE FUNCTION inc2(val integer) RETURNS integer AS $$
return val + 2;
$$
LANGUAGE pldotnet;
SELECT inc2(8) = integer '10';

CREATE OR REPLACE FUNCTION sum2(a integer, b integer) RETURNS integer AS $$
return a + b;
$$
LANGUAGE pldotnet;
SELECT sum2(3,2) = integer '5';

CREATE OR REPLACE FUNCTION sum3(aaa integer, bbb integer, ccc integer) RETURNS integer AS $$
return aaa + bbb + ccc;
$$
LANGUAGE pldotnet;
SELECT sum3(3,2,1) = integer '6';

CREATE OR REPLACE FUNCTION sum4(a integer, b integer, c integer, d integer) RETURNS integer AS $$
return a + b + c + d;
$$
LANGUAGE pldotnet;
SELECT sum4(4,3,2,1) = integer '10';


