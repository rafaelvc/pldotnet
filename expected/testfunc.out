CREATE OR REPLACE FUNCTION returnX() RETURNS integer AS $$
return 10;
$$ LANGUAGE pldotnet;
SELECT returnX();
 returnx 
---------
      10
(1 row)

CREATE OR REPLACE FUNCTION inc2(val integer) RETURNS integer AS $$
return val + 2;
$$
LANGUAGE pldotnet;
SELECT inc2(8);
 inc2 
------
   10
(1 row)

CREATE OR REPLACE FUNCTION sum2(a integer, b integer) RETURNS integer AS $$
return a + b;
$$
LANGUAGE pldotnet;
SELECT sum2(3,2);
 sum2 
------
    5
(1 row)

CREATE OR REPLACE FUNCTION sum3(aaa integer, bbb integer, ccc integer) RETURNS integer AS $$
return aaa + bbb + ccc;
$$
LANGUAGE pldotnet;
SELECT sum3(3,2,1);
 sum3 
------
    6
(1 row)

CREATE OR REPLACE FUNCTION sum4(a integer, b integer, c integer, d integer) RETURNS integer AS $$
return a + b + c + d;
$$
LANGUAGE pldotnet;
SELECT sum4(4,3,2,1);
 sum4 
------
   10
(1 row)

